clients/ssh/git_upload_pack.go

```diff
@@ -0,0 +1,223 @@
+package ssh
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/url"
+	"os"
+
+	"gopkg.in/src-d/go-git.v2/clients/common"
+	"gopkg.in/src-d/go-git.v2/formats/pktline"
+
+	"github.com/sourcegraph/go-vcsurl"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+)
+
+var (
+	InvalidAuthMethodErr      = errors.New("invalid ssh auth method: a ssh.SSHAuthMethod should be provided.")
+	AuthRequiredErr           = errors.New("cannot connect: auth required.")
+	NotConnectedErr           = errors.New("not connected")
+	AlreadyConnectedErr       = errors.New("already connected")
+	UploadPackAnswerFormatErr = errors.New("git-upload-pack bad answer format")
+	UnsupportedVCSErr         = errors.New("only git is supported")
+	UnsupportedRepoErr        = errors.New("only github.com is supported")
```

Comment:
Minor thing related to Go style, could we move the Err on all of these to the beginning? Examples:
https://golang.org/pkg/os/#pkg-variables, https://golang.org/pkg/net/http/#pkg-variables